Sources/CodableKit/Reflection/ReflectionDecodable.swift

```diff
@@ -94,42 +105,76 @@ extension ReflectionDecodable where Self: Equatable {
 extension String: ReflectionDecodable {
     /// See `ReflectionDecodable`.
     public static func reflectDecoded() -> (String, String) { return ("0", "1") }
+
+    public static var isBaseType: Bool { return true }
 }
 
 extension FixedWidthInteger {
     /// See `ReflectionDecodable`.
     public static func reflectDecoded() -> (Self, Self) { return (0, 1) }
+
+    public static var isBaseType: Bool { return true }
 }
 
-extension UInt: ReflectionDecodable { }
-extension UInt8: ReflectionDecodable { }
-extension UInt16: ReflectionDecodable { }
-extension UInt32: ReflectionDecodable { }
-extension UInt64: ReflectionDecodable { }
+extension UInt: ReflectionDecodable {
+    public static var isBaseType: Bool { return true }
```

Comment:
`isBaseType` should default to `true` so that we don't need to specify it for all of these types.

Reply:
I think it makes more sense to call this `isPrimitiveType` and have the default value be `false` with overrides in the types handled by Vapor (See https://github.com/vapor/core/pull/191).

This saves everyone else using this library in having to specify that property. I can't see any case where I'd make my custom type conforming to `ReflectDecodable` and wanting `isBaseType = true`.

Reply:
I agree @valeriomazzeo, with the new name and the behaviour. However, I am need to set all the types manually then, since Swift complains when setting a default type on `FixedWidthInteger`. So @tanner0101, should I add the lines back again for this behaviour?